user_sync/rules.py

```diff
@@ -447,7 +465,14 @@ def add_dashboard_user(self, user_key, dashboard_connectors):
             owning_organization_info = self.get_organization_info(OWNING_ORGANIZATION_NAME)        
             desired_groups = owning_organization_info.get_desired_groups(user_key)
             groups_to_add = self.calculate_groups_to_add(owning_organization_info, user_key, desired_groups)
-            commands.add_groups(groups_to_add)
+
```

Comment:
I actually meant to have helper to help you add groups_to_add into commands.   Inside the helper, you would do the by_type to get subset and call commands.add.  Because commands.add_groups is still there, you can sort of make use of that.   

Like designations = [DESIGNATION_GROUP, DESTIGNATION_PRODUCT];
for each desgination, do by_type and do add_groups(subset, type).